### PR TITLE
Quasar 17.15.0 plat 23219

### PR DESCRIPTION
**Pull request by danbar0kal**

### CODE REVIEW

File: alpha/lib/model/UserLoginDataPeer.php [hunk 1]
```diff
@@ -727,7 +758,7 @@ public static function addLoginData($loginEmail, $password, $partnerId, $firstNa
 			throw new kUserException('', kUserException::INVALID_PARTNER);
 		}
 		
-		if ($isAdminUser)
+		if ($isAdminUser && strpos($userRoleNames, $partner->getLimitedAdminRoleName()) === false)
```
Comment (inbal-ben-david) [inline alpha/lib/model/UserLoginDataPeer.php:761]: explode the user role names and check if the excluded role name is in_array

File: alpha/lib/model/UserLoginDataPeer.php [hunk 2]
```diff
@@ -716,7 +747,7 @@ private static function findFirstValidKuser($loginDataId, $notPartnerId = null)
 	 * @throws kUserException::LOGIN_ID_ALREADY_USED
 	 * @throws kUserException::ADMIN_LOGIN_USERS_QUOTA_EXCEEDED
 	 */
-	public static function addLoginData($loginEmail, $password, $partnerId, $firstName, $lastName, $isAdminUser, $checkPasswordStructure = true, &$alreadyExisted = null)
+	public static function addLoginData($loginEmail, $password, $partnerId, $firstName, $lastName, $isAdminUser, $checkPasswordStructure = true, &$alreadyExisted = null, $userRoleNames)
```
Comment (inbal-ben-david) [inline alpha/lib/model/UserLoginDataPeer.php:750]: add default value to userRoleNames

File: alpha/lib/model/kuserPeer.php [hunk 3]
```diff
@@ -10,18 +10,18 @@
 class kuserPeer extends BasekuserPeer implements IRelatedObjectPeer
 {	
 	const KALTURA_NEW_USER_EMAIL = 120;
```
Comment (inbal-ben-david) [inline alpha/lib/model/kuserPeer.php:12]: align this one as well

File: alpha/lib/model/kuserPeer.php [hunk 4]
```diff
@@ -571,25 +571,34 @@ public static function sendNewUserMail(kuser $user, $existingUser)
 	{
 		// setup parameters
 		$partnerId = $user->getPartnerId();
+		$partner = PartnerPeer::retrieveByPK($partnerId);
+		$platformUserRole = $partner->getLimitedAdminRoleName();
```
Comment (inbal-ben-david) [inline alpha/lib/model/kuserPeer.php:575]: you are not using this anymore

File: alpha/lib/model/kuserPeer.php [hunk 5]
```diff
@@ -617,30 +628,82 @@ public static function sendNewUserMail(kuser $user, $existingUser)
...
+					kEmails::TAG_PARTNER_ID            => $partnerId,
+					kEmails::TAG_ROLE_NAME             => $roleName,
+					kEmails::TAG_PUSER_ID              => $puserId,
+					kEmails::TAG_KMC_LINK              => $kmcLink,
+					kEmails::TAG_CONTACT_LINK          => $contactLink,
+					kEmails::TAG_BEGINNERS_GUID_LINK   => $beginnersGuideLink,
+					kEmails::TAG_QUICK_START_GUID_LINK => $quickStartGuideLink);
+				if ($authType == PartnerAuthenticationType::SSO)
+				{
+					$associativeBodyParams += [kEmails::TAG_LOGIN_LINK => $bodyParams[3]];
```
Comment (inbal-ben-david) [inline alpha/lib/model/kuserPeer.php:674]: this is not clear

File: api_v3/lib/types/batch/KalturaMailJobData.php [hunk 6]
```diff
@@ -54,7 +54,12 @@ class KalturaMailJobData extends KalturaJobData
 	/**
 	 * @var string
 	 */    
-    public $subjectParams;  
+    public $subjectParams;
```
Comment (inbal-ben-david) [inline api_v3/lib/types/batch/KalturaMailJobData.php:57]: indentation

**Review by inbal-ben-david — COMMENTED** [review]
need to also add release notes

File: alpha/lib/model/kuserPeer.php [hunk 7]
```diff
@@ -617,30 +628,82 @@ public static function sendNewUserMail(kuser $user, $existingUser)
 				$mailType = self::KALTURA_NEW_USER_ADMIN_CONSOLE_EMAIL;
 				$bodyParams = array($userName, $creatorUserName, $loginEmail, $resetPasswordLink, $roleName, $adminConsoleLink, $qrCodeLink);
 			}
+			if ($isDynamicEmailRole)
```
Comment (inbal-ben-david) [inline alpha/lib/model/kuserPeer.php:631]: change to a flag named useDynamicEmailTemplate

File: infra/general/kEmails.php [hunk 8]
```diff
@@ -5,7 +5,85 @@
...
+			$dynamicEmailContents->setEmailSubject($dynamicSubjects[$mailType]);
+			return $dynamicEmailContents;
+		}
+		else
+		{
+			throw new KalturaAPIException(KalturaEmailNotificationErrors::DYNAMIC_EMAIL_CONTENT_TEMPLATE_FAULT);
+		}
+	}
+	
+	public static function getUseDynamicEmailTemplate($userRoleNames =  null)
```
Comment (inbal-ben-david) [inline infra/general/kEmails.php:57]: fix spaces
change function name to getDynamicEmailUserRoleNmae

File: alpha/lib/model/UserLoginDataPeer.php [hunk 9]
```diff
@@ -410,6 +422,26 @@ public static function getPassResetLink($hashKey, $linkType = resetPassLinkType:
...
 	
+	protected static function getResetLinkPrefix($partnerId, $linkType, $dynamicLink = null)
+	{
+		if ($dynamicLink)
+		{
+			return $dynamicLink;
+		}
+		if($linkType == resetPassLinkType::KMS)
+		{
+			$resetLinksArray = kConf::get('password_reset_links');
```
Comment (inbal-ben-david) [inline alpha/lib/model/UserLoginDataPeer.php:433]: can extract it once before the if/else

File: alpha/lib/model/UserLoginDataPeer.php [hunk 10]
```diff
@@ -410,6 +422,26 @@ public static function getPassResetLink($hashKey, $linkType = resetPassLinkType:
...
+			$resetLinkPrefix = $resetLinksArray['kms'];
+			$resetLinkPrefix = vsprintf($resetLinkPrefix, array($partnerId) );
```
Comment (inbal-ben-david) [inline alpha/lib/model/UserLoginDataPeer.php:435]: fix spaces

File: alpha/lib/model/UserLoginDataPeer.php [hunk 11]
```diff
@@ -701,6 +733,11 @@ private static function findFirstValidKuser($loginDataId, $notPartnerId = null)
 		return null;
 	}
 	
+	protected static function isLimitedAdminRole($partner, $userRoleNames)
```
Comment (inbal-ben-david) [inline alpha/lib/model/UserLoginDataPeer.php:736]: change to excluded

File: alpha/lib/model/kuserPeer.php [hunk 12]
```diff
@@ -571,25 +571,33 @@ public static function sendNewUserMail(kuser $user, $existingUser)
...
-		$partner = PartnerPeer::retrieveByPK($partnerId);
+		
 		$publisherName = $partner->getName();
 		$loginEmail = $user->getEmail();
-		$roleName = $user->getUserRoleNames();
+		$roleNames = $user->getUserRoleNames();
 		$puserId = $user->getPuserId();
-		if (!$existingUser) {
-			$resetPasswordLink = UserLoginDataPeer::getPassResetLink($user->getLoginData()->getPasswordHashKey());
+		$useDynamicEmailTemplate = kEmails::getUseDynamicEmailTemplate($roleNames);
```
Comment (inbal-ben-david) [inline alpha/lib/model/kuserPeer.php:594]: change var name to match user role name meanning

Comment (danbar0kal) [reply alpha/lib/model/kuserPeer.php:594]: $roleNameToUseDynamicEmailTemplate

**Pull request merged**